[pt][quant] Update the misleading comments for zero_points and scale in dynamic quant linear module

jianyuh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28827 [pt][quant] Avoid the misleading zero_point and scale [2/2]
* **#28767 [pt][quant] Update the misleading comments for zero_points and scale in dynamic quant linear module**

The scale and zero_point are for the output activation tensor, not for the weight tensor;

Differential Revision: [D18164949](https://our.internmc.facebook.com/intern/diff/D18164949/)

jianyuh committed:
[pt][quant] Update the misleading comments for zero_points and scale in dynamic quant linear module

The scale and zero_point are for the output activation tensor, not for the weight tensor;

Differential Revision: [D18164949](https://our.internmc.facebook.com/intern/diff/D18164949/)

[ghstack-poisoned]

z-a-f commented on torch/nn/quantized/dynamic/modules/linear.py:
```diff
@@ -17,8 +17,8 @@ class Linear(nnq.Linear):
-        scale: `scale` parameter of weight Quantized Tensor, type: double
-        zero_point: `zero_point` parameter for weight Quantized Tensor, type: long
+        scale: `scale` parameter of output activation Quantized Tensor, type: double
+        zero_point: `zero_point` parameter for output activation Quantized Tensor, type: long
```
There is no reason to capitalize the "Quantized Tensor" -- let's keep it as "quantized tensor"

z-a-f reviewed (approved):
LGTM

raghuramank100 reviewed (changes requested):
I think this comment is misleading, please see my comments below

raghuramank100 commented on torch/nn/quantized/dynamic/modules/linear.py:
```diff
@@ -17,8 +17,8 @@ class Linear(nnq.Linear):
                 If :attr:`bias` is ``True``, the values are initialized to zero.
-        scale: `scale` parameter of weight Quantized Tensor, type: double
-        zero_point: `zero_point` parameter for weight Quantized Tensor, type: long
+        scale: `scale` parameter of output activation Quantized Tensor, type: double
```
For dynamic quantization there is no output scale and zero-point. We should not be exposing these in the comments. The activations are in floating point.

jianyuh replied:
We have this PR, mainly because when we print the modules, we print the scale and zero points:
https://github.com/pytorch/pytorch/blob/master/torch/nn/quantized/modules/linear.py#L47-L48

The example output is for the RoBERTa model after dynamic quantization:
```
      (19): TransformerEncoderLayer(
        (dropout): Dropout(p=0.1, inplace=False)
        (attention): MultiheadAttention(
          (dropout): Dropout(p=0.1, inplace=False)
          (input_projection): DynamicQuantizedLinear(in_features=1024, out_features=3072, scale=1.0, zero_point=0)
          (output_projection): DynamicQuantizedLinear(in_features=1024, out_features=1024, scale=1.0, zero_point=0)
        )
        (residual_mlp): ResidualMLP(
          (mlp): Sequential(
            (0): DynamicQuantizedLinear(in_features=1024, out_features=4096, scale=1.0, zero_point=0)
            (1): GeLU()
            (2): Dropout(p=0.1, inplace=False)
            (3): DynamicQuantizedLinear(in_features=4096, out_features=1024, scale=1.0, zero_point=0)
            (4): Dropout(p=0.1, inplace=False)
          )
        )
        (attention_layer_norm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
        (final_layer_norm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
      )
      (20): TransformerEncoderLayer(
        (dropout): Dropout(p=0.1, inplace=False)
        (attention): MultiheadAttention(
          (dropout): Dropout(p=0.1, inplace=False)
          (input_projection): DynamicQuantizedLinear(in_features=1024, out_features=3072, scale=1.0, zero_point=0)
          (output_projection): DynamicQuantizedLinear(in_features=1024, out_features=1024, scale=1.0, zero_point=0)
        )
        (residual_mlp): ResidualMLP(
          (mlp): Sequential(
            (0): DynamicQuantizedLinear(in_features=1024, out_features=4096, scale=1.0, zero_point=0)
            (1): GeLU()
            (2): Dropout(p=0.1, inplace=False)
            (3): DynamicQuantizedLinear(in_features=4096, out_features=1024, scale=1.0, zero_point=0)
            (4): Dropout(p=0.1, inplace=False)
          )
        )
        (attention_layer_norm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
        (final_layer_norm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
      )
```

In the comment, we would like to tell the users that the `scale=1.0` and `zero_point=0` for `DynamicQuantizedLinear` module are for the output activation (to be consistent with the static quantization).

jianyuh committed:
Update on "[pt][quant] Update the misleading comments for zero_points and scale in dynamic quant linear module"


The scale and zero_point are for the output activation tensor, not for the weight tensor;

Differential Revision: [D18164949](https://our.internmc.facebook.com/intern/diff/D18164949/)

[ghstack-poisoned]

jianyuh commented:
Per @raghuramank100 's request, we removed the comments for `scale` and `zero_points`, and added the `extra_repr` in https://github.com/pytorch/pytorch/pull/28827.
